src/structures/Match.js

```diff
@@ -0,0 +1,354 @@
+     */
+    get participantType(){
+        return this._data.participanttype;
+    }
+
+    /**
+     * The match script settings
+     * NOTE: Array is empty (api update?)
+     * @type {Array<MatchScriptSetting>}
+     * @private
+     */
+    get scriptSettings(){
+        return this._data.scriptsettings;
+    }
+
+    /**
+     * The match maps
+     * NOTE: Array is empty (api update?)
+     * @type {Array<MatchMap>}
+     * @private
+     */
+    get maps(){
+        return this._data.maps;
+    }
+
+    /**
+     * The match server id
+     * @type {number}
+     */
+    get serverId(){
+        return this._data.serverid;
+    }
+
+    /**
+     * The match join link
+     * @type {string}
+     */
+    get joinLink(){
+        return this._data.serverjoinlink;
+    }
+
+    /**
+     * The match status
+     * @type {MatchStatus}
+     */
+    get status(){
+        return this._data.status;
+    }
+
+    /**
+     * The match players
+     * @type {Array<MatchPlayer>}
+     */
+    get players(){
+        return this._data.players.map(player => new MatchPlayer(this, player));
+    }
+
+    /**
+     * The match teams (if match is completed)
+     * @type {?Array<MatchTeam>}
+     */
+    get teams(){
+        if(this.status === "COMPLETED"){
+            if (this.type == MMTypes[3]) {
+                // In royal, there are 20 teams but the API returns 30 items, so we need to delete the last 10 teams
+                return this._data.teams.slice(0, 20).map(team => new MatchTeam(this, team));
+            }
+            return this._data.teams.map(team => new MatchTeam(this, team));
+        }
+        return null;
+    }
+}
+
+/**
+ * The team in the match
+ */
+class MatchTeam {
+    constructor(match, data){
+        /**
+         * The match
+         * @type {Match}
+         */
+        this.match = match;
+
+        /**
+         * The client instance.
+         * @type {Client}
+         */
+        this.client = match.client;
+
+        /**
+         * The data
+         * @type {Object}
+         * @private
+         */
+        this._data = data;
+    }
+
+    /**
+     * The team Index
+     * @type {number}
+     */
+    get index(){
+        return this._data.index;
+    }
+
+    /**
+     * The team score
+     * @type {number}
+     */
+    get score(){
+        return this._data.score;
+    }
+
+    /**
+     * The team name
+     * @type {?string}
+     */
+    get name(){
+        if (this.match.type == MMTypes[2]){ // 3v3
+            if (this.index == 0) return "Blue";
+            if (this.index == 1) return "Red";
+        } else if (this.match.type == MMTypes[3]){ // Royal
+            return RoyalTeams[this.index].name;
+        }
+        return null;
+    }
+
+    /**
+     * The team image (if the match is Royal)
+     * @type {?string}
+     */
+    get image(){
+        if (this.match.type == MMTypes[3]){ // Royal
+            return RoyalTeams[this.index].img;
+        }
+        else return null;
```

Comment:
Obviously this is linter stuff, but I just noticed it so I thought I'd mention it 🙂 This `else` doesn't do anything.